### ISSUE TITLE
pubsub.subscribe option parameter not optional in 0.27

### ISSUE THREAD

**Issue by kevinsimper** [ipfs/js-ipfs#1129]
According to the docs, option is optional, but it is required in 0.27 as it will try to call the callback as the third parameter.
https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/PUBSUB.md#pubsubsubscribe

If only two parameters were given, both callback and handler would be undefined

https://github.com/ipfs/js-ipfs/blob/bd264f27ba2f435c3ae3388c7503d6118627bd73/src/core/components/pubsub.js#L16

**Comment by diasdavid** [comment]
@kevinsimper Wanna submit a test that shows that incorrect behavior?

**Comment by achingbrain** [comment]
If two arguments are passed, `callback` and `handler` would indeed be undefined.  The block in question reassigns `options` to `handler` if `options` is a function and then assigns `handler` to `callback` (to cover the case where you've passed three arguments).  Later on we guard on `callback` and return a promise if it's falsey.

So if there are two args it should be ok as it'll just result in a promise being returned from the function call, which is what happens [in this test case](https://github.com/ipfs/interface-ipfs-core/blob/master/js/src/pubsub.js#L166).

@kevinsimper have you observed any unexpected behaviour?

**Comment by kevinsimper** [comment]
It has been fixed in the newest version of ipfs `0.27.7` and I just tested again and got the error still in `0.27.0`, so good 👍 

https://jsbin.com/zuvuxijuma/edit?html,js,output
```
var id = 'ipfs-repo-' + String(Math.random())
console.log({id})
const node = new Ipfs({
  repo: id,
  EXPERIMENTAL: {
    pubsub: true
  },
  config: {
    Addresses: {
      Swarm: [
        '/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star'
      ]
    }
  }
})

node.on('ready', function() {
  // Your node is now ready to use \o/
  console.log('ready')
  // stopping a node

  let topic = 'fruit-of-the-daykevin'

  let receiveMsg = function(msg) {
    console.log(msg.data.toString(), Date.now())
  }

  node.pubsub.subscribe(topic, receiveMsg)
  setInterval(function() {
    node.pubsub.peers(topic, (err, peerIds) => {
      if (err) {
        throw err
      }
      console.log(peerIds)
    })
  }, 1000)
  setInterval(function() {
    node.pubsub.publish(topic, new node.types.Buffer('apple ' + id + ' ' + Date.now()), function(
      e
    ) {
      console.log({ e })
    })
  }, 4000)
})

```

This will fail:
```
  <script src="https://cdn.jsdelivr.net/npm/ipfs@0.27.0/dist/index.js"></script>
```
with this error:
```
pubsub.js:27 Uncaught TypeError: callback is not a function
    at setImmediate (pubsub.js:27)
    at setImmediate.js:27
    at run (setImmediate.js:40)
    at runIfPresent (setImmediate.js:69)
    at onGlobalMessage (setImmediate.js:109)
```

However works fine now with:
```
<script src="https://cdn.jsdelivr.net/npm/ipfs@0.27.7/dist/index.js"></script>
```

**Comment by diasdavid** [comment]
👍 great, thanks for letting us know @kevinsimper
